internal/broker/testdata/TestIsAuthenticated/golden/authenticating_with_password_refreshes_expired_token/first_call

```diff
@@ -1,3 +1,3 @@
 access: granted
-data: '{"userinfo":{"name":"test-user@email.com","uuid":"test-user-id","dir":"/home/test-user@email.com","shell":"/usr/bin/bash","gecos":"test-user@email.com","groups":[{"name":"remote-group","ugid":"12345"},{"name":"linux-local-group","ugid":""}]}}'
+data: '{"userinfo":{"name":"test-user@email.com","uuid":"saved-user-id","dir":"/home/test-user@email.com","shell":"/usr/bin/bash","gecos":"test-user@email.com","groups":[{"name":"saved-remote-group","ugid":"12345"},{"name":"saved-local-group","ugid":""}]}}'
```

Comment:
Again a diff that shouldn't exist, no?

Reply:
The change comes from the merge commit where I merged main to get the revert of 30d23aca2baedb29b3c0c60e87d8a300c0ff50b4. The tests fail if I undo the change. I want to do a git bisect to find out which change caused the tests to produce a different output. In order to do that, I'm rebasing on main to have a linear git history which I can bisect.

Reply:
So the reason for the diff in the golden file is that `broker.loadAuthInfo` used to not load the `UserInfo` field from the cache, only the , and the `Token` and `RawIDToken` fields. The new `token.LoadAuthInfo` implementation also loads the `UserInfo` field. Do you know why the old implementation didn't set that field?

Reply:
If we refresh the token, it means that user information could also have changed so we need to fetch it again. The only cases where we rely on the user information in the cache are:
1) Access token is still valid, so we don't refresh anything (it will change once we go back to handling this part)
2) We are in an offline session, so we can't fetch the user information from the provider.

Reply:
I got rid of the diff by always refreshing the user info exactly then when we also refresh the token (by moving the `fetchUserInfo` call inside the `if !authInfo.Token.Valid() && !session.isOffline` block in https://github.com/ubuntu/authd-oidc-brokers/blob/aec153c9576b5f0367a883d8232b813d52d39368/internal/broker/broker.go#L557-L562). 

That's not the behavior we want, but we already track that in https://github.com/ubuntu/authd/issues/520, which we shouldn't fix in this PR.